src/main/java/com/yahoo/bullet/storm/JoinBolt.java

```diff
@@ -344,10 +340,11 @@ private void startDelayed(Map.Entry<String, Querier> query) {
 
     private Metadata withSignal(Metadata metadata, Metadata.Signal signal) {
         // Don't change the non-readonly bits of metadata in place since that might affect tuples emitted but pending.
-        Metadata copy = new Metadata(signal, null);
-        if (metadata != null) {
-            copy.setContent(metadata.getContent());
+        if (metadata == null) {
+            return new Metadata(signal, null);
```

Comment:
Isn't this gonna just crash the PublisherBolt instance anyway when the PubSub publisher in there tries to cast?

Reply:
If the metadata is null, it will only be sent to the feedback stream (loop bolt) which uses query publisher and not response publisher (so it accepts metadata and wraps it).